python/lsst/pipe/base/pipelineIR.py

```diff
@@ -874,17 +946,12 @@ def from_uri(cls, uri: Union[str, ButlerURI]) -> PipelineIR:
             The loaded pipeline
         """
         loaded_uri = ButlerURI(uri)
-        # With ButlerURI we have the choice of always using a local file or
-        # reading in the bytes directly. Reading in bytes can be more
-        # efficient for reasonably-sized files when the resource is remote.
-        # For now use the local file variant. For a local file as_local() does
-        # nothing.
-        with loaded_uri.as_local() as local:
+        with loaded_uri.open("r") as buffer:
             # explicitly read here, there was some issue with yaml trying
             # to read the ButlerURI itself (I think because it only
             # pretends to be conformant to the io api)
-            loaded_yaml = yaml.load(local.read(), Loader=PipelineYamlLoader)
-            return cls(loaded_yaml)
+            loaded_yaml = yaml.load(buffer, Loader=PipelineYamlLoader)
```

Comment:
That _sounds_ like the missing functionality https://jira.lsstcorp.org/browse/DM-32842 added, but I also don't understand how it could have even kind of worked before.  Anyhow, I think the situation is sufficiently different after that ticket that it's worth a try, and I'll remove the comment.